AppCenterData/AppCenterData/MSData.m

```diff
@@ -776,6 +782,14 @@ - (void)networkStateChanged:(NSNotificationCenter *)__unused notification {
 }
 
 - (void)processPendingOperations {
+
+  // Only process pending operations when auth context is available.
+  NSNumber *authContextFlag = [MS_USER_DEFAULTS objectForKey:kMSAuthContextAvailabilityKey];
+  if (!authContextFlag.boolValue) {
```

Comment:
What happens if a user is authenticated but the token is expired after a while? This will still lead 404 error?

Reply:
the token will be renewed explicitly. this is not any different from the current behavior.

Reply:
It is probably an edge case but if it takes time to refresh token but Data started and trying to talk to token exchange service. It still has a chance to get 404 in that case?

Reply:
when Data wants to manipulate doc, it needs a token, if the cached token is expired, then it tries to get a new token.
if it fails to renew a token, then it’ll log error which is a valid case.
the bug was that Data tried to get token when app is not signed in, therefore, it logged 404